src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts

```diff
@@ -95,31 +97,41 @@ export class MLSConversationVerificationStateHandler {
    * This function returns the WireIdentity of all userDeviceEntities in a conversation, as long as they have a certificate.
    * If the conversation has userDeviceEntities without a certificate, it will not be included in the returned array
    *
+   * It also updates the isMLSVerified observable of all the devices in the conversation
    */
-  private getAllUserEntitiesInConversation = async (conversation: Conversation) => {
-    if (!conversation.groupId) {
-      this.logger.error('Conversation has no groupId', conversation.name);
-      throw new Error('Conversation has no groupId');
-    }
-
+  private updateUserDevices = async (conversation: Conversation) => {
     const userEntities = conversation.getAllUserEntities();
-
-    const deviceUserPairs = userEntities
-      .flatMap(userEntity => {
-        return userEntity.devices().map(device => ({[device.id]: userEntity.qualifiedId}));
-      })
-      .reduce((acc, current) => {
-        return {...acc, ...current};
-      }, {});
-
-    const identities = await this.core.service!.e2eIdentity!.getUserDeviceEntities(
-      conversation.groupId,
-      deviceUserPairs,
-    );
+    const allClients: ClientEntity[] = [];
+    const allIdentities: WireIdentity[] = [];
+    userEntities.forEach(async userEntity => {
+      const devices = userEntity.devices();
+      const deviceUserPairs = devices
+        .map(device => ({
+          [device.id]: userEntity.qualifiedId,
+        }))
+        .reduce((acc, current) => {
+          return {...acc, ...current};
+        }, {});
+      const identities = await this.core.service!.e2eIdentity!.getUserDeviceEntities(
+        conversation.groupId!,
```

Comment:
I think there is a `isMLSConversation` that will also make sure that this `groupId` is present ;)

Reply:
have a look at the `ConversationSelector.ts`